### PR TITLE
Add python file support

### LINKED ISSUE #9: Python file support
This extension is great! Just wanted to suggest a feature: instead of going blank when editing a Python file, it could show a skeleton of the code with the class/method lines.

### PR DESCRIPTION
**Pull request by marthacryan**
Fixes #9. Adds a table of contents for .py files, with classes and methods displayed. The classes are displayed in a larger font than the methods: 
image

### CODE REVIEW

**Comment by lresende** [comment]
Thanks @marthacryan to getting this up so quickly. 

Overall it looks good, but I have a few suggestions:
- Should we add an outline item if we find imports? This might be useful to help dependency management (e.g. validating a package has been imported, or import new ones)
- Similar to what we have in the Markdown, should we have some sort of collapse/expand capability, so, you can click on a class and it would toggle the list of methods. This would be useful in large python scripts that define multiple classes.

@sballin do you want to give this a try and provide any other additional feedback?

**Comment by kgryte** [comment]
@marthacryan Nice! Thanks for maintaining the code style!

@lresende Collapse/expand functionality could be added in a future PR. I don't think this should be a stopper for this current PR.

**Comment by kgryte** [comment]
My only other comment is that the implemented approach is "naive" in the sense that corner cases are not considered. For example, something like

```python
"""
Definition is defined as a statement of meaning for a term.
"""
```

If we wanted to be pedantic, we'd need to parse a Python file into an AST.

However, until such corner cases are flagged as issues, I think the current approach is suitable as a first pass, providing functionality which can be leveraged to gather community feedback.

**Comment by marthacryan** [comment]
Just added an imports section to toc. Also removed the ":" at the end of each entry on the toc.

**Comment by marthacryan** [comment]
Realized I was cutting off the last character of the import entries. Here's a screenshot with imports:
image

File: src/generators/python/index.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  // Iterate over the lines to get the heading level and text for each line:
+  let headings: IHeading[] = [];
+  let processingImports = false;
+  for (let i = 0; i < lines.length; i++) {
+    let line = lines[i].trimLeft();
+    if (line.indexOf('def ') === 0) {
+      processingImports = false;
+      headings.push({
+        text: line.slice(0, -1),
```
Comment (lresende) [inline src/generators/python/index.ts]: Above you are doing a trimLeft, which means that if there are trailing spaces on the right then `:` will not be handled properly.

**Comment by lresende** [comment]
@kgryte any other suggestions, comments for this first pass? Otherwise looks good to me.

**Comment by kgryte** [comment]
LGTM

**Pull request merged**